server/src/main/java/org/elasticsearch/search/aggregations/support/CoreValuesSourceType.java

```diff
@@ -310,8 +310,17 @@ public ValuesSource getField(FieldContext fieldContext, AggregationScript.LeafFa
                         }
                     }
 
-                    // Check the query for bounds
-                    if (context.query() != null) {
+                    boolean isMultiValue;
+                    try {
+                        isMultiValue = context.isMultiValued(fieldContext.field());
+                    } catch (AggregationExecutionException ex) {
+                        // If we can't check, don't agressively disable the optimization
+                        isMultiValue = false;
```

Comment:
Do we hit this case when the field is declared but is empty?

Reply:
I suppose if the field isn't searchable the query wouldn't find anything. But it might just not be indexed. Or empty. I guess. I'm not sure `false` is right here - but maybe it doesn't matter because we never get here. If that's the case, can we just throw?